store/tikv/region_cache.go

```diff
@@ -961,19 +999,20 @@ func (c *RegionCache) switchToPeer(r *Region, targetStoreID uint64) (found bool)
 
 func (c *RegionCache) switchNextPeer(r *Region, currentPeerIdx int, err error) {
 	rs := r.getStore()
-	if int(rs.workStoreIdx) != currentPeerIdx {
```

Comment:
If we move the check after increment `store.fail`, we may increment `fail` more than we need.
@lysu Can it cause any problem?

Reply:
The check is still there, it only guards updating of region store. We use this to detect failures when doing follower read.